File: RQ2_Prompt_Elements/OpenAI_Data/HumanEvalPython_output/scenario2/id_124T2000Test.py
# id_124_Test.java
from scenario2.id_124 import valid_date

import pytest

class id_124_Test:
    """
    Test class of id_124.py.
    It contains ten unit test cases for the :func:`scenario2.id_124.valid_date`.
    """


    def test_valid_date_1(self):
        """
        Test case 1 for :func:`scenario2.id_124.valid_date`.
        """
        assert valid_date('01-01-2000') == True

    def test_valid_date_2(self):
        """
        Test case 2 for :func:`scenario2.id_124.valid_date`.
        """
        assert valid_date('02-29-2000') == True

    def test_valid_date_3(self):
        """
        Test case 3 for :func:`scenario2.id_124.valid_date`.
        """
        assert valid_date('02-29-2001') == False

    def test_valid_date_4(self):
        """
        Test case 4 for :func:`scenario2.id_124.valid_date`.
        """
        assert valid_date('02-30-2000') == False

    def test_valid_date_5(self):
        """
        Test case 5 for :func:`scenario2.id_124.valid_date`.
        """
        assert valid_date('02-31-2000') == False

    def test_valid_date_6(self):
        """
        Test case 6 for :func:`scenario2.id_124.valid_date`.
        """
        assert valid_date('04-31-2000') == False

    def test_valid_date_7(self):
        """
        Test case 7 for :func:`scenario2.id_124.valid_date`.
        """
        assert valid_date('06-31-2000') == False

    def test_valid_date_8(self):
        """
        Test case 8 for :func:`scenario2.id_124.valid_date`.
        """
        assert valid_date('09-31-2000') == False

    def test_valid_date_9(self):
        """
        Test case 9 for :func:`scenario2.id_124.valid_date`.
        """
        assert valid_date('11-31-2000') == False

    def test_valid_date_10(self):
        """
        Test case 10 for :func:`scenario2.id_124.valid_date`.
        """
        assert valid_date('13-31-2000') == False

File: RQ1_Test_Generation/StarCoder_Data/HumanEvalJava_output/original/id_124T2000Test.java
// ValidDateTest.java
package original;


import org.junit.jupiter.api.Test;
import static org.junit.jupiter.api.Assertions.*;

/**
* Test class of {@link ValidDate}.
* It contains ten unit test cases for the {@link ValidDate#validDate(String)} method.
*/
class ValidDateTest {

    /**
    * Test case for {@link ValidDate#validDate(String)}
    * when the date string is not empty.
    */
    @Test
    void testValidDate1() {
        assertTrue(ValidDate.validDate("03-11-2000"));
    }

    /**
    * Test case for {@link ValidDate#validDate(String)}
    * when the number of days is not less than 1 or higher than 31 days for months 1,3,5,7,8,10,12.
    */
    @Test
    void testValidDate2() {
        assertFalse(ValidDate.validDate("15-01-2012"));
    }

    /**
    * Test case for {@link ValidDate#validDate(String)}
    * when the number of days is not less than 1 or higher than 30 days for months 4,6,9,11.
    */
    @Test
    void testValidDate3() {
        assertFalse(ValidDate.validDate("04-0-2040"));
    }

    /**
    * Test case for {@link ValidDate#validDate(String)}
    * when the number of days is not less than 1 or higher than 29 for the month 2.
    */
    @Test
    void testValidDate4() {
        assertTrue(ValidDate.validDate("06-04-2020"));
    }

    /**
    * Test case for {@link ValidDate#validDate(String)}
    * when the months should not be less than 1 or higher than 12.
    */
    @Test
    void testValidDate5() {
        assertFalse(ValidDate.validDate("06/04/2020"));
    }

    /**
    * Test case for {@link ValidDate#validDate(String)}
    * when the date should be in the format: mm-dd-yyyy.
    */
    @Test
    void testValidDate6() {
        assertTrue(ValidDate.validDate("01-01-2007"));
    }

    /**
    * Test case for {@link ValidDate#validDate(String)}
    * when the date is not empty.
    */
    @Test
    void testValidDate7() {
        assertFalse(ValidDate.validDate("03-32-2011"));
    }

    /**
    * Test case for {@link ValidDate#validDate(String)}
    * when the date is empty.
    */
    @Test
    void testValidDate8() {
        assertFalse(ValidDate.validDate(""));
    }

    /**
    * Test case for {@link ValidDate#validDate(String)}
    * when the date is not empty.
    */
    @Test
    void testValidDate9() {
        assertFalse(ValidDate.validDate("04-31-3000"));
    }

    /**
    * Test case for {@link ValidDate#validDate(String)}
    * when the date is not empty.
    */
    @Test
    void testValidDate10() {
        assertTrue(ValidDate.validDate("06-06-2005"));
    }
}
